src/mainloop.cpp

```diff
@@ -557,7 +523,7 @@ void Mainloop::_del_timeouts()
     }
 }
 
-void Mainloop::_add_tcp_retry(TcpEndpoint *tcp)
+void Mainloop::add_tcp_retry(TcpEndpoint *tcp)
```

Comment:
Now that our code I much cleaner, we probably want to remove this from Mainloop. No need to keep the Timeout in the Mainloop rather than TcpEndpoint

Reply:
Just this method (which will call `Mainloop::get_instance().add_timeout()` then) or or the whole timeout stuff?

Reply:
Just implemented this as well. Because `add_timeout()` is also used by the mainloop itself, I just moved the all of the reconnect handling to the TCP endpoint. The `valid` flag now indicates, that the endpoint can be removed.